hera_cal/smooth_cal.py

```diff
@@ -350,9 +351,102 @@ def rephase_to_refant(gains, refant, flags=None, propagate_refant_flags=False):
                 gains[ant] = gains[ant] / refant_phasor
 
 
+def build_time_blacklist(time_grid, time_blacklists=[], lst_blacklists=[], lat_lon_alt_degrees=None, telescope_name='HERA'):
+    '''Converts pairs of bounds on blacklisted times/LSTs into a boolean array of blacklisted times.
+
+    Arguments:
+        time_grid: numpy array of times in Julian Day
+        time_blacklists: list of pairs of times in Julian Day bounding (inclusively) regions in time that are to be marked
+            as True in the time_blacklist_array
+        lst_blacklists:  list of pairs of LSTs in hours bounding (inclusively) regions of LST that are to be marked as True
+            in the time_blacklist_array. Regions crossing the 24h brach cut, e.g. [(23, 1)], are allowed.
+        lat_lon_alt_degrees: length 3 array of telescope location in degreees and altitude in meters. Only used to convert
+            times to LSTs if lst_blacklists is not empty
+        telescope_name: string name of telescope. Only used if lst_blacklists is not empty and lat_lon_alt_degrees 
+            is not None. Currently, only "HERA" will work since it's position is hard-coded in this module.
+
+    Returns:
+        time_blacklist_array: boolean array with the same shape as time_grid in which blacklisted integrations are True'''
+    
+    time_blacklist_array = np.zeros(len(time_grid), dtype=bool)
+
+    # Calculate blacklisted times
+    if len(time_blacklists) > 0:
+        for bounds in time_blacklists:
+            assert isinstance(bounds, Iterable) and len(bounds) == 2, 'time_blacklists must be list of pairs of bounds'
+            assert bounds[0] <= bounds[1], 'time_blacklist bounds must be in chronological order'
```

Comment:
We could lift the restriction from the user to input the time in chronological by doing this evaluation in the code itself.

Reply:
I think it'd be quite odd for the user to put the constraints in the opposite order; I think it's more likely that the user put the times out of order by mistake and I'd rather the code not fail silently. 

Also, if we don't enforce chronological order, then specifying 23-1 for LST blacklists becomes ambiguous whether you want to blacklist a total of 2 hours or a total of 22 hours.

So I'd prefer not to make this change, but I'm open to making the error message more clear if you have suggestions.

Reply:
Thanks. Okay, I think I agree with you here. Then for consistency better to enforce the time, frequency and channel inputs to be in chronological order except the lst since the lst ranges from 24-0 hours.